src/main/java/be/seeseemelk/mockbukkit/block/state/AbstractFurnaceMock.java

```diff
@@ -0,0 +1,154 @@
+package be.seeseemelk.mockbukkit.block.state;
+
+import be.seeseemelk.mockbukkit.UnimplementedOperationException;
+import be.seeseemelk.mockbukkit.inventory.InventoryMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.Block;
+import org.bukkit.block.Furnace;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.CookingRecipe;
+import org.bukkit.inventory.FurnaceInventory;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+public abstract class AbstractFurnaceMock extends ContainerMock implements Furnace
+{
+
+	private short burnTime;
+	private short cookTime;
+	private int cookTimeTotal;
+	private double cookSpeedMultiplier = 1.0;
+
+	protected AbstractFurnaceMock(@NotNull Material material)
+	{
+		super(material);
+		if (material != Material.SMOKER && material != Material.FURNACE && material != Material.BLAST_FURNACE)
+			throw new IllegalArgumentException("Cannot create a Furnace state from " + material);
+	}
+
+	protected AbstractFurnaceMock(@NotNull Block block)
+	{
+		super(block);
+		if (block.getType() != Material.SMOKER && block.getType() != Material.FURNACE && block.getType() != Material.BLAST_FURNACE)
+			throw new IllegalArgumentException("Cannot create a Furnace state from " + block.getType().name());
+	}
+
+	protected AbstractFurnaceMock(@NotNull ContainerMock state)
+	{
+		super(state);
+	}
```

Comment:
Shouldn't we do some Type Checking here too?

Reply:
Wrong parameter type, fixes